submission/file-sources/src/main/kotlin/uk/ac/ebi/io/sources/FtpSource.kt

```diff
@@ -22,15 +21,10 @@ class FtpSource(
     }
 
     private fun findFile(filePath: String): File? {
-        val ftpPath = basePath.resolve(filePath)
+        val ftpPath = ftpPath.resolve(filePath)
         val files = ftpClient.listFiles(ftpPath)
         if (files.isEmpty()) return null
-
-        val fileName = filePath.substringAfterLast("/")
-        val target = Files.createTempFile("ftp-file", fileName)
-
-        target.outputStream().use { ftpClient.downloadFile(ftpPath, it) }
-        return target.toFile()
+        return nfsPath.resolve(filePath).toFile()
```

Comment:
If I understood correctly, this is what we're doing:
1. We use the ftp client to list the files
2. If the file is not found we return null
3. If it exists, we return whatever is on `nfsPath.resolve(filePath)`

If we're returning the file from the nfsPath and the only reason to need the ftp client is to list the files, wouldn't this be equivalent to:

```
    val file = nfsPath.resolve(filePath)
    return if (file.exists()) file.toFile() else null
```

I just want to understand the need for the ftp client here